From ea82fc151c5c092afb08a72289252c359ac88f45 Mon Sep 17 00:00:00 2001
From: Bingyu Zhou <rain.by.zhou@gmail.com>
Date: Wed, 19 Apr 2017 19:47:41 +0800
Subject: [PATCH 36/37] staging: rtl8723bs: Fix assignment in if condition in
 os_dep/sdio_intf.c

ERROR: do not use assignment in if condition
Detected by scripts/checkpatch.pl -f

Signed-off-by: Bingyu Zhou <rain.by.zhou@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/rtl8723bs/os_dep/sdio_intf.c | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

--- a/drivers/staging/rtl8723bs/os_dep/sdio_intf.c
+++ b/drivers/staging/rtl8723bs/os_dep/sdio_intf.c
@@ -258,7 +258,8 @@ static struct dvobj_priv *sdio_dvobj_ini
 	struct dvobj_priv *dvobj = NULL;
 	PSDIO_DATA psdio;
 
-	if ((dvobj = devobj_init()) == NULL) {
+	dvobj = devobj_init();
+	if (dvobj == NULL) {
 		goto exit;
 	}
 
@@ -336,7 +337,8 @@ static struct adapter *rtw_sdio_if1_init
 	struct adapter *padapter = NULL;
 	PSDIO_DATA psdio = &dvobj->intf_data;
 
-	if ((padapter = (struct adapter *)vzalloc(sizeof(*padapter))) == NULL) {
+	padapter = (struct adapter *)vzalloc(sizeof(*padapter));
+	if (padapter == NULL) {
 		goto exit;
 	}
 
@@ -481,18 +483,21 @@ static int rtw_drv_init(
 	struct adapter *if1 = NULL, *if2 = NULL;
 	struct dvobj_priv *dvobj;
 
-	if ((dvobj = sdio_dvobj_init(func)) == NULL) {
+	dvobj = sdio_dvobj_init(func);
+	if (dvobj == NULL) {
 		RT_TRACE(_module_hci_intfs_c_, _drv_err_, ("initialize device object priv Failed!\n"));
 		goto exit;
 	}
 
-	if ((if1 = rtw_sdio_if1_init(dvobj, id)) == NULL) {
+	if1 = rtw_sdio_if1_init(dvobj, id);
+	if (if1 == NULL) {
 		DBG_871X("rtw_init_primarystruct adapter Failed!\n");
 		goto free_dvobj;
 	}
 
 	/* dev_alloc_name && register_netdev */
-	if ((status = rtw_drv_register_netdev(if1)) != _SUCCESS) {
+	status = rtw_drv_register_netdev(if1);
+	if (status != _SUCCESS) {
 		goto free_if2;
 	}
 
